201303开始 www.szse.cn
201804开始 docs.static.szse.cn
https://akshare.akfamily.xyz/data/stock/stock.html#id9

In [20]:
import akshare as ak
import pandas as pd
import requests

from akshare.utils.cons import headers

In [24]:
ak.stock_industry_category_cninfo(symbol="全球行业分类标准")

,类目编码,类目名称,终止日期,行业类型,行业类型编码,类目名称英文,父类编码,分级
0,10,能源,NaT,全球行业分类标准（GICS）,008008,None,Y,1
1,1010,能源,NaT,全球行业分类标准（GICS）,008008,None,10,2
2,101010,能源设备与服务,NaT,全球行业分类标准（GICS）,008008,None,1010,3
3,10101010,石油与天然气钻井,NaT,全球行业分类标准（GICS）,008008,None,101010,4
4,10101020,石油天然气设备与服务,NaT,全球行业分类标准（GICS）,008008,None,101010,4
...,...,...,...,...,...,...,...,...
302,60201010,多元化房地产业务,NaT,全球行业分类标准（GICS）,008008,None,602010,4
303,60201020,房地产经营公司,NaT,全球行业分类标准（GICS）,008008,None,602010,4
304,60201030,房地产开发,NaT,全球行业分类标准（GICS）,008008,Real Estate Development,602010,4
305,60201040,房地产服务,NaT,全球行业分类标准（GICS）,008008,None,602010,4


In [34]:
ak.stock_industry_change_cninfo(symbol="000001", start_date="20200101", end_date="20250808")

,新证券简称,行业中类,行业大类,行业次类,行业门类,机构名称,行业编码,分类标准,分类标准编码,证券代码,变更日期
0,平安银行,综合性银行,商业银行,银行,金融地产,平安银行股份有限公司,06010101,中证行业分类标准(旧),008019,000001,2020-03-16
1,平安银行,股份制银行Ⅲ,股份制银行Ⅲ,股份制银行Ⅱ,银行,平安银行股份有限公司,S480301,申银万国行业分类标准,008003,000001,2021-07-30
2,平安银行,综合性银行,商业银行,银行,金融,平安银行股份有限公司,40101010,中证行业分类标准,008014,000001,2021-12-17
3,平安银行,NaN,货币金融服务,NaN,金融业,平安银行股份有限公司,J66,中国上市公司协会上市公司行业分类标准,008001,000001,2024-02-08


In [36]:
ak.stock_share_change_cninfo(symbol="002594", start_date="20091227", end_date="20251021")

,证券简称,机构名称,境外法人持股,证券投资基金持股,国家持股-受限,国有法人持股,配售法人股,发起人股份,未流通股份,其中：境外自然人持股,...,总股本,其中：限售高管股,转配股,境内上市外资股-B股,其中：境外法人持股,募集法人股,已流通股份,其中：境内自然人持股,其他内资持股-受限,变动原因编码
0,比亚迪,比亚迪股份有限公司,NaN,0.0,0.0,NaN,1500.0,0.0,0.0,0.0,...,235410.0000,NaN,NaN,0.0,1742.604,NaN,85710.0000,129518.2500,146457.3960,"015001,015019"
1,比亚迪,比亚迪股份有限公司,NaN,0.0,0.0,NaN,0.0,0.0,0.0,0.0,...,235410.0000,NaN,NaN,0.0,1742.604,NaN,87210.0000,129518.2500,146457.3960,015009
2,比亚迪,比亚迪股份有限公司,NaN,0.0,0.0,NaN,0.0,NaN,0.0,0.0,...,235410.0000,NaN,NaN,0.0,1742.604,NaN,87210.0000,129518.2500,146457.3960,015019
3,比亚迪,比亚迪股份有限公司,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,235410.0000,NaN,NaN,NaN,NaN,NaN,114884.5950,103909.7150,120525.4050,015019
4,比亚迪,比亚迪股份有限公司,NaN,0.0,0.0,NaN,0.0,0.0,0.0,0.0,...,235410.0000,NaN,NaN,0.0,1742.604,NaN,134930.0200,82121.6860,98737.3760,015027
5,比亚迪,比亚迪股份有限公司,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,235410.0000,NaN,NaN,NaN,NaN,NaN,113759.4515,105034.8585,121650.5485,015019
6,比亚迪,比亚迪股份有限公司,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,235410.0000,NaN,NaN,NaN,NaN,NaN,117737.2435,101057.0665,117672.7565,015019
7,比亚迪,比亚迪股份有限公司,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,235410.0000,NaN,NaN,NaN,NaN,NaN,117737.2435,101057.0665,117672.7565,015019
8,比亚迪,比亚迪股份有限公司,NaN,NaN,NaN,NaN,0.0,0.0,0.0,NaN,...,247600.0000,NaN,NaN,0.0,NaN,NaN,129927.2435,101057.0665,117672.7565,015005
9,比亚迪,比亚迪股份有限公司,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,247600.0000,0.0,NaN,NaN,NaN,NaN,160764.9735,74641.3870,86835.0265,015019


In [41]:
ak.stock_profile_cninfo(symbol="600288")

,公司名称,英文名称,曾用简称,A股代码,A股简称,B股代码,B股简称,H股代码,H股简称,入选指数,...,官方网站,电子邮箱,联系电话,传真,注册地址,办公地址,邮政编码,主营业务,经营范围,机构简介
0,大恒新纪元科技股份有限公司,"Daheng New Epoch Technology, Inc.",None,600288,大恒科技,None,None,None,None,国证Ａ指,...,www.dhxjy.com.cn,600288@dhxjy.com.cn,010-82827855,010-82827853,北京市海淀区苏州街3号大恒科技大厦北座13层,北京市海淀区苏州街3号大恒科技大厦北座15层,100080,以光机电一体化产品和信息技术产品为主要内容的高新技术产品开发、生产和经营。,光学、激光、红外元器件及设备、计算机软硬件、多媒体产品、办公自动化设备、医疗器械、精细化工及...,于1998年由原中国新纪元物资流通中心（现已更名为中国新纪元有限公司）作为主发起人，联合中国...


In [ ]:
ak.tool_trade_date_hist_sina()

## 股票市场总貌
### 上海证券交易所
* 目标地址: http://www.sse.com.cn/market/stockdata/statistic/
* 描述: 上海证券交易所-股票数据总貌
* 限量: 单次返回最近交易日的股票数据总貌(当前交易日的数据需要交易所收盘后统计)

In [25]:
ak.stock_sse_summary()

,项目,股票,主板,科创板
0,流通股本,47540.86,45725.36,1815.5
1,总市值,579751.08,502125.94,77625.15
2,平均市盈率,14.85,13.66,54.93
3,上市公司,2285,1696,589
4,上市股票,2323,1734,589
5,流通市值,543506.68,480987.04,62519.64
6,报告时间,20250730,20250730,20250730
8,总股本,50600.04,48283.34,2316.71


### 上海证券交易所-每日概况
* 目标地址: http://www.sse.com.cn/market/stockdata/overview/day/
* 描述: 上海证券交易所-数据-股票数据-成交概况-股票成交概况-每日股票情况
* 限量: 单次返回指定日期的每日概况数据, 当前交易日数据需要在收盘后获取; 注意仅支持获取在 20211227（包含）之后的数据

In [27]:
ak.stock_sse_deal_daily(date="20250730")

,单日情况,股票,主板A,主板B,科创板,股票回购
0,挂牌数,2323.0000,1693.0000,41.000,589.0000,0.00
1,市价总值,579751.0800,501155.7800,970.150,77625.1500,0.00
2,流通市值,543506.6800,480266.7800,720.250,62519.6400,0.00
3,成交金额,8205.8600,6571.8300,1.870,1632.1500,4.71
4,成交量,637.8900,591.7300,0.400,45.7600,0.31
5,平均市盈率,14.8500,13.6700,9.940,54.9300,NaN
6,换手率,1.4154,1.3113,0.193,2.1026,0.00
7,流通换手率,1.5098,1.3684,0.260,2.6106,0.00


### 深圳证券交易所
* 目标地址: http://www.szse.cn/market/overview/index.html
* 描述: 深圳证券交易所-市场总貌-证券类别统计
* 限量: 单次返回指定 date 的市场总貌数据-证券类别统计(当前交易日的数据需要交易所收盘后统计)

In [ ]:
ak.stock_szse_summary(date="20250730")

### 股票行业成交
* 目标地址: http://docs.static.szse.cn/www/market/periodical/month/W020220511355248518608.html
* 描述: 深圳证券交易所-统计资料-股票行业成交数据
* 限量: 单次返回指定 symbol 和 date 的统计资料-股票行业成交数据

In [20]:
ak.stock_szse_sector_summary(symbol="当年", date="202402")

/home/ts/.local/share/virtualenvs/jupyter.12-csASsZfE/lib/python3.12/site-packages/bs4/__init__.py:228: UserWarning: You provided Unicode markup but also provided a value for from_encoding. Your from_encoding will be ignored.
  warnings.warn("You provided Unicode markup but also provided a value for from_encoding. Your from_encoding will be ignored.")


,项目名称,项目名称-英文,交易天数,成交金额-人民币元,成交金额-占总计,成交股数-股数,成交股数-占总计,成交笔数-笔,成交笔数-占总计
0,合计,Total,37,16519610425683,100.00,1645989602341,100.00,1809844098,100.00
1,农林牧渔,Agriculture,37,119705225615,0.72,11946897890,0.73,13570091,0.75
2,采矿业,Mining,37,219246759921,1.33,21890572421,1.33,25135148,1.39
3,制造业,Manufacturing,37,10836209287810,65.60,957940705778,58.20,1183263485,65.38
4,水电煤气,Utilities,37,231037264303,1.40,40501722258,2.46,36570885,2.02
5,建筑业,Construction,37,266289270422,1.61,42098783217,2.56,31070054,1.72
6,批发零售,Wholesale & Retail,37,457392004067,2.77,66048738616,4.01,58824685,3.25
7,运输仓储,Transportation,37,134466581993,0.81,16955218455,1.03,19660803,1.09
8,住宿餐饮,Hotels & Catering,37,37997133621,0.23,5471118764,0.33,4927101,0.27
9,信息技术,IT,37,2215546625829,13.41,196447289602,11.93,220458142,12.18


In [23]:
df1 = ak.stock_szse_sector_summary(symbol="当年", date="202506").drop('项目名称-英文',axis=1)
df2 = ak.stock_szse_sector_summary(symbol="当年", date="202406").drop('项目名称-英文',axis=1)

/home/ts/.local/share/virtualenvs/jupyter.12-csASsZfE/lib/python3.12/site-packages/bs4/__init__.py:228: UserWarning: You provided Unicode markup but also provided a value for from_encoding. Your from_encoding will be ignored.
  warnings.warn("You provided Unicode markup but also provided a value for from_encoding. Your from_encoding will be ignored.")
/home/ts/.local/share/virtualenvs/jupyter.12-csASsZfE/lib/python3.12/site-packages/bs4/__init__.py:228: UserWarning: You provided Unicode markup but also provided a value for from_encoding. Your from_encoding will be ignored.
  warnings.warn("You provided Unicode markup but also provided a value for from_encoding. Your from_encoding will be ignored.")


In [24]:
((((df1['成交金额-人民币元']/df1['交易天数']))/100000000)-(((df2['成交金额-人民币元']/df2['交易天数']))/100000000)).round(2)

0     3385.11
1       25.66
2        0.18
3     2185.93
4       11.68
5      -12.64
6      137.56
7       18.09
8        0.86
9      725.22
10     124.03
11      -3.67
12      89.99
13      12.35
14      24.21
15      -0.21
16       2.25
17      17.10
18      21.79
19       4.72
dtype: float64

In [ ]:
ak.stock_szse_sector_summary(symbol="当月", date="201301").drop('项目名称-英文',axis=1)

In [ ]:
ak.stock_szse_sector_summary(symbol="当月", date="201701").drop('项目名称-英文',axis=1)

In [ ]:
import pandas as pd
import requests
from bs4 import BeautifulSoup
from io import BytesIO, StringIO

初始化需要修改代码，2018年4月后数据不用修改

In [ ]:
def stock_szse_sector_summary(
    symbol: str = "当月", date: str = "202501"
) -> pd.DataFrame:
    """
    深圳证券交易所-统计资料-股票行业成交数据
    https://docs.static.szse.cn/www/market/periodical/month/W020220511355248518608.html
    :param symbol: choice of {"当月", "当年"}
    :type symbol: str
    :param date: 交易年月
    :type date: str
    :return: 股票行业成交数据
    :rtype: pandas.DataFrame
    """
    url = "https://www.szse.cn/market/periodical/month/index.html"
    r = requests.get(url)
    r.encoding = "utf8"
    soup = BeautifulSoup(r.text, features="lxml")
    tags_list = soup.find_all(name="div", attrs={"class": "g-container"})[1].find_all(
        "script"
    )
    tags_dict = [
        eval(
            item.string[item.string.find("{") : item.string.find("}") + 1]
            .replace("\n", "")
            .replace(" ", "")
            .replace("value", "'value'")
            .replace("text", "'text'")
        )
        for item in tags_list
    ]
    date_url_dict = dict(
        zip(
            [item["text"] for item in tags_dict],
            [item["value"][2:] for item in tags_dict],
        )
    )
    date_format = "-".join([date[:4], date[4:]])
    url = f"https://www.szse.cn/market/periodical/month/{date_url_dict[date_format]}"
    r = requests.get(url)
    r.encoding = "utf8"
    soup = BeautifulSoup(r.text, features="lxml")
    url = [
        item for item in soup.find_all("a") if item.get_text() == "股票行业成交数据"
    ][0]["href"]

    if 'http' in url :
        pass
    else:
        url = 'https://www.szse.cn'+url

    if symbol == "当月":
        r = requests.get(url)
        r.encoding = 'GBK'
        temp_df = pd.read_html(StringIO(r.text), encoding="gbk")[0]
        temp_df.columns = [
            "项目名称",
            "项目名称-英文",
            "交易天数",
            "成交金额-人民币元",
            "成交金额-占总计",
            "成交股数-股数",
            "成交股数-占总计",
            "成交笔数-笔",
            "成交笔数-占总计",
        ]
    else:
        temp_df = pd.read_html(url, encoding="gbk")[1]
        temp_df.columns = [
            "项目名称",
            "项目名称-英文",
            "交易天数",
            "成交金额-人民币元",
            "成交金额-占总计",
            "成交股数-股数",
            "成交股数-占总计",
            "成交笔数-笔",
            "成交笔数-占总计",
        ]

    temp_df["交易天数"] = pd.to_numeric(temp_df["交易天数"], errors="coerce")
    temp_df["成交金额-人民币元"] = pd.to_numeric(
        temp_df["成交金额-人民币元"], errors="coerce"
    )
    temp_df["成交金额-占总计"] = pd.to_numeric(
        temp_df["成交金额-占总计"], errors="coerce"
    )
    temp_df["成交股数-股数"] = pd.to_numeric(temp_df["成交股数-股数"], errors="coerce")
    temp_df["成交股数-占总计"] = pd.to_numeric(
        temp_df["成交股数-占总计"], errors="coerce"
    )
    temp_df["成交笔数-笔"] = pd.to_numeric(temp_df["成交笔数-笔"], errors="coerce")
    temp_df["成交笔数-占总计"] = pd.to_numeric(
        temp_df["成交笔数-占总计"], errors="coerce"
    )
    return temp_df

In [ ]:
stock_szse_sector_summary(symbol="当月", date="201501")

In [ ]:
gen = (f"{year}{month:02d}" for year in range(2000,2101) for month in range(1,13))

In [ ]:
# 生成2010-2015年所有月份的序列 
ylist = [f"{year}{month:02d}" for year in range(2013, 2025) for month in range(1,13)] + ['202501']

In [ ]:
ddf = []

In [ ]:
for n in ylist:
    df = ak.stock_szse_sector_summary(symbol="当月", date= n ).drop('项目名称-英文',axis=1)
    df['日期'] = n
    ddf.append(df)



In [ ]:
dff = pd.concat(ddf)

In [ ]:
dff.set_index('日期')

In [ ]:
from sqlalchemy import create_engine
eng = create_engine('postgresql+psycopg://sa:11111111@10.3.18.56:5432/StockBas')

In [ ]:
dff.set_index('日期').to_sql('szSum', eng, if_exists="replace")

In [ ]:
ddf = pd.DataFrame(columns=[ '日期','项目名称', '交易天数', '成交金额-人民币元', '成交金额-占总计', '成交股数-股数', '成交股数-占总计',
       '成交笔数-笔', '成交笔数-占总计'],dtype='object')

In [ ]:
n = ylist[0]
df = ak.stock_szse_sector_summary(symbol="当月", date= n ).drop('项目名称-英文',axis=1)
df['日期'] = n
# ddf.append(df)
ddf = pd.concat([ddf, df],axis=0,ignore_index=True,copy=False)


### 个股信息查询
#### 1、东财
* 目标地址: http://quote.eastmoney.com/concept/sh603777.html?from=classic
* 描述: 东方财富-个股-股票信息
* 限量: 单次返回指定 symbol 的个股信息

In [28]:
ak.stock_individual_info_em(symbol="600996")

,item,value
0,最新,8.13
1,股票代码,600996
2,股票简称,贵广网络
3,总股本,1246799854.0
4,流通股,1246799854.0
5,总市值,10136482813.02
6,流通市值,10136482813.02
7,行业,文化传媒
8,上市时间,20161226


行情报价

In [29]:
ak.stock_bid_ask_em(symbol="600996")

,item,value
0,sell_5,8.17
1,sell_5_vol,56500.00
2,sell_4,8.16
3,sell_4_vol,28300.00
4,sell_3,8.15
5,sell_3_vol,55000.00
6,sell_2,8.14
7,sell_2_vol,78500.00
8,sell_1,8.13
9,sell_1_vol,15600.00


### 主营介绍
同花顺
* 目标地址: https://basic.10jqka.com.cn/new/000066/operate.html
* 限量: 单次返回所有数据

In [32]:
ak.stock_zyjs_ths(symbol="000066")

,股票代码,主营业务,产品类型,产品名称,经营范围
0,000066,计算产业、系统装备及其他业务。,计算产业、系统装备,计算产业、系统装备,计算机软件、硬件、终端及其外部设备、网络系统及系统集成、电子产品及零部件、金融机具、税控机具...


#### 主营构成-东财
* 目标地址: https://emweb.securities.eastmoney.com/PC_HSF10/BusinessAnalysis/Index?type=web&code=SH688041#
* 描述: 东方财富网-个股-主营构成
* 限量: 单次返回所有历史数据

In [34]:
ak.stock_zygc_em(symbol="SZ000066")

,股票代码,报告日期,分类类型,主营构成,主营收入,收入比例,主营成本,成本比例,主营利润,利润比例,毛利率
0,000066,2024-12-31,NaN,计算产业,1.070153e+10,0.753483,9.138324e+09,0.760125,1.563204e+09,0.716860,0.146073
1,000066,2024-12-31,NaN,系统装备,2.683068e+09,0.188912,2.312758e+09,0.192375,3.703103e+08,0.169818,0.138017
2,000066,2024-12-31,NaN,其他,8.181567e+08,0.057606,5.710450e+08,0.047499,2.471117e+08,0.113321,0.302035
3,000066,2024-12-31,按产品分类,计算产业,1.070153e+10,0.753483,9.138324e+09,0.760125,1.563204e+09,0.716860,0.146073
4,000066,2024-12-31,按产品分类,系统装备,2.683068e+09,0.188912,2.312758e+09,0.192375,3.703103e+08,0.169818,0.138017
...,...,...,...,...,...,...,...,...,...,...,...
195,000066,2019-06-30,按产品分类,园区及物业服务,3.171339e+07,0.007324,1.721776e+07,0.005057,1.449564e+07,0.015661,0.457082
196,000066,2019-06-30,按产品分类,其他,1.066304e+06,0.000246,1.552112e+06,0.000456,-4.858084e+05,-0.000525,-0.455600
197,000066,2019-06-30,按地区分类,国内,3.226267e+09,0.745056,2.466162e+09,0.724354,7.601056e+08,0.821204,0.235599
198,000066,2019-06-30,按地区分类,其他(补充),4.305256e+08,0.099423,3.604920e+08,0.105883,7.003362e+07,0.075663,0.162670


实时行情数据-东财 沪深京A stock_zh_a_spot_em()

沪A股  _sh_
        深A股 _sz_
        京A股 _bj_
        新股 _new_  
        创业板 _cy_  
        科创板 _kc_



In [ ]:
ak.stock_zh_a_spot_em()

 单次返回指定股票最近一个交易日的分时数据, 包含盘前数据

In [ ]:
ak.stock_intraday_em(symbol="000409")

In [ ]:
ak.stock_intraday_em(symbol="000409").groupby('买卖盘性质').sum()

目标地址: https://data.eastmoney.com/gsrl/gsdt.html
公司动态

In [ ]:
ak.stock_gsrl_gsdt_em(date="20250211")

沪深新股
目标地址: https://quote.eastmoney.com/center/gridlist.html#newshares

In [ ]:
ak.stock_zh_a_new_em()

机构调研http://data.eastmoney.com/jgdy/tj.html
http://data.eastmoney.com/jgdy/xx.html

单次返回所有历史数据

In [ ]:
ak.stock_jgdy_tj_em(date="20210128")

In [ ]:
ak.stock_jgdy_detail_em(date="20241211")

In [ ]:
ak.stock_zyjs_ths(symbol="600996")

In [ ]:
ak.stock_zygc_em(symbol="SH600996")

In [ ]:
ak.stock_zygc_ym(symbol="000001")

上市公司质押比例
目标地址: https://data.eastmoney.com/gpzy/pledgeRatio.aspx 查询具体交易日

重要股东股权质押明细 https://data.eastmoney.com/gpzy/pledgeDetail.aspx 


In [ ]:
ak.stock_gpzy_pledge_ratio_em(date="20250207")

In [ ]:
ak.stock_gpzy_pledge_ratio_detail_em()

In [ ]:
def calculate_alert_liquidation_price(initial_price, pledge_rate, interest_rate, alert_ratio=1.6, liquidation_ratio=1.4):
    """
    initial_price: 初始股价（元）
    pledge_rate: 质押率（0-1）
    interest_rate: 融资利率（年化，0-1）
    alert_ratio: 预警线系数（默认160%）
    liquidation_ratio: 平仓线系数（默认140%）
    """
    factor = initial_price * pledge_rate * (1 + interest_rate)
    alert_price = factor * alert_ratio
    liquidation_price = factor * liquidation_ratio
    return alert_price, liquidation_price 

In [ ]:
calculate_alert_liquidation_price(8.2, 0.4999, 0.06)

个股商誉明细 目标地址: https://data.eastmoney.com/sy/list.html

行业商誉 目标地址: https://data.eastmoney.com/sy/hylist.html


In [ ]:
ak.stock_sy_em(date="20221231")


In [ ]:
ak.stock_sy_hy_em(date="20231231")

分析师指数 目标地址: https://data.eastmoney.com/invest/invest/list.html
 分析师详情 https://data.eastmoney.com/invest/invest/11000257131.html

In [ ]:
ak.stock_analyst_rank_em() 

In [ ]:
ak.stock_analyst_detail_em(analyst_id="11000280036", indicator="最新跟踪成分股")

千股千评 目标地址: https://data.eastmoney.com/stockcomment/

In [ ]:
ak.stock_comment_em()

In [ ]:
ak.stock_comment_detail_zlkp_jgcyd_em(symbol="600996")

In [ ]:
ak.stock_comment_detail_zhpj_lspf_em(symbol="600996")

In [ ]:
ak.stock_comment_detail_scrd_focus_em(symbol="600996")

In [ ]:
ak.stock_comment_detail_scrd_desire_daily_em(symbol="600996")

停复牌信息

In [ ]:
ak.stock_tfp_em()

个股新闻 目标地址: https://so.eastmoney.com/news/s
财经内容精选 目标地址: https://cxdata.caixin.com/pc/

In [ ]:
ak.stock_news_em(symbol="600996")

In [ ]:
ak.stock_news_main_cx()

年报季报 目标地址: http://data.eastmoney.com/bbsj/202412/yjbb.html

date="20200331"; choice of {"0331一季报", "0630中报", "0930三季报", "1231年报"}; 从 20100331 开始


In [ ]:
ak.stock_yjbb_em(date="20241231")

In [ ]:
ak.stock_yjkb_em(date="20241231")

In [ ]:
ak.stock_yjyg_em(date="20241231")

预约披露时间-东方财富

In [ ]:
ak.stock_yysj_em(symbol="沪深A股", date="20241231")

预约披露时间-巨潮资讯 目标地址: http://www.cninfo.com.cn/new/commonUrl?url=data/yypl

period="2021年报"; 近四期的财务报告; e.g., choice of {"2021一季", "2021半年报", "2021三季", "2021年报"}


In [ ]:
ak.stock_report_disclosure(market="沪深京", period="2024年报")

上市公司行业归属的变动情况-巨潮资讯

In [ ]:
ak.stock_industry_change_cninfo(symbol="002594", start_date="20050101", end_date="20250212")

公司股本变动-巨潮资讯

In [ ]:
fd = ak.stock_share_change_cninfo(symbol="600996", start_date="20050101", end_date="20250212")

In [ ]:
df = ak.stock_profile_cninfo(symbol="600996")

高管持股

In [ ]:
ak.stock_ggcg_em(symbol="全部")

分红配送

In [ ]:
ak.stock_fhps_em(date="20241231")

In [ ]:
ak.stock_fhps_detail_em(symbol="600409")

In [ ]:
ak.stock_fhps_detail_ths(symbol="600409")

资金流向 

symbol="即时"; choice of {“即时”, "3日排行", "5日排行", "10日排行", "20日排行"}

个股资金流：individual  概念资金流：concept 行业资金流：industry 

In [ ]:
ak.stock_fund_flow_individual(symbol="即时")

筹码分布 adjust=""; choice of {"qfq": "前复权", "hfq": "后复权", "": "不复权"}

In [ ]:
ak.stock_cyq_em(symbol="600996", adjust="")

个股研报

In [ ]:
ak.stock_research_report_em(symbol="600409")


股东户数

In [ ]:
ak.stock_zh_a_gdhs(symbol="20241231")

In [ ]:
ak.stock_zh_a_gdhs_detail_em(symbol="600996")

限售股解禁详情

In [ ]:
ak.stock_restricted_release_detail_em(start_date="20240326", end_date="20250212")

股票列表-A股 目标地址: 沪深京三个交易所

In [35]:
ak.stock_info_a_code_name()

  0%|          | 0/14 [00:00<?, ?it/s]

,code,name
0,000001,平安银行
1,000002,万 科Ａ
2,000004,*ST国华
3,000006,深振业Ａ
4,000007,全新好
...,...,...
5414,920445,龙竹科技
5415,920489,佳先股份
5416,920682,球冠电缆
5417,920799,艾融软件


symbol="主板A股"; choice of {"主板A股", "主板B股", "科创板"}


In [36]:
ak.stock_info_sh_name_code(symbol="主板A股")

,证券代码,证券简称,公司全称,上市日期
0,600000,浦发银行,上海浦东发展银行股份有限公司,1999-11-10
1,600004,白云机场,广州白云国际机场股份有限公司,2003-04-28
2,600006,东风股份,东风汽车股份有限公司,1999-07-27
3,600007,中国国贸,中国国际贸易中心股份有限公司,1999-03-12
4,600008,首创环保,北京首创生态环保集团股份有限公司,2000-04-27
...,...,...,...,...
1688,605580,恒盛能源,恒盛能源股份有限公司,2021-08-19
1689,605588,冠石科技,南京冠石科技股份有限公司,2021-08-12
1690,605589,圣泉集团,济南圣泉集团股份有限公司,2021-08-10
1691,605598,上海港湾,上海港湾基础建设(集团)股份有限公司,2021-09-17


symbol="A股列表"; choice of {"A股列表", "B股列表", "CDR列表", "AB股列表"}

In [37]:
ak.stock_info_sz_name_code(symbol="A股列表")

,板块,A股代码,A股简称,A股上市日期,A股总股本,A股流通股本,所属行业
0,主板,000001,平安银行,1991-04-03,"19,405,918,198","19,405,600,653",J 金融业
1,主板,000002,万 科Ａ,1991-01-29,"9,724,196,533","9,716,483,304",K 房地产
2,主板,000004,*ST国华,1990-12-01,"132,380,282","126,287,768",I 信息技术
3,主板,000006,深振业Ａ,1992-04-27,"1,349,995,046","1,349,987,396",K 房地产
4,主板,000007,全新好,1992-04-13,"346,448,044","346,448,044",K 房地产
...,...,...,...,...,...,...,...
2863,创业板,301658,首航新能,2025-04-02,"412,371,135","38,728,009",C 制造业
2864,创业板,301662,宏工科技,2025-04-17,"80,000,000","15,776,325",C 制造业
2865,创业板,301665,泰禾股份,2025-04-11,"450,000,000","35,906,289",C 制造业
2866,创业板,301678,新恒汇,2025-06-20,"239,555,467","45,512,773",C 制造业


In [38]:
ak.stock_info_bj_name_code()

,证券代码,证券简称,总股本,流通股本,上市日期,所属行业,地区,报告日期
0,430017,星昊医药,122288200,121417900,2023-05-31,医药制造业,北京市,2025-07-31
1,430047,诺思兰德,274271974,180106427,2020-11-24,医药制造业,北京市,2025-07-31
2,430090,同辉信息,199333546,135347400,2021-08-09,软件和信息技术服务业,北京市,2025-07-31
3,430139,华岭股份,269507392,260468480,2022-10-28,计算机、通信和其他电子设备制造业,上海市,2025-07-31
4,430198,微创光电,161363872,70152938,2020-07-27,计算机、通信和其他电子设备制造业,湖北省,2025-07-31
...,...,...,...,...,...,...,...,...
264,920445,龙竹科技,146930025,105262791,2020-07-27,家具制造业,福建省,2025-07-31
265,920489,佳先股份,136459200,85577249,2020-07-27,化学原料和化学制品制造业,安徽省,2025-07-31
266,920682,球冠电缆,324480000,188324422,2020-07-27,电气机械和器材制造业,浙江省,2025-07-31
267,920799,艾融软件,209780050,120704571,2020-07-27,软件和信息技术服务业,上海市,2025-07-31


### 暂停/终止上市
* 上证
* symbol="全部"; choice of {"全部", "沪市", "科创板"}

In [41]:
ak.stock_info_sh_delist(symbol="全部")

,公司代码,公司简称,上市日期,暂停上市日期
0,600001,邯郸钢铁,1998-01-22,2009-12-29
1,600002,齐鲁退市,1998-04-08,2006-04-24
2,600003,ST东北高,1999-08-10,2010-02-26
3,600005,武钢股份,1999-08-03,2017-02-14
4,600065,*ST联 谊,1997-05-23,2007-12-13
...,...,...,...,...
141,900950,新城Ｂ股,1997-10-16,2015-11-23
142,900951,退市大化,1997-10-21,2020-08-28
143,900953,凯马Ｂ,1998-06-24,2024-07-05
144,900956,东贝Ｂ股,1999-07-15,2020-11-23


#### 终止/暂停上市
* 深证
* symbol="终止上市公司"; choice of {"暂停上市公司", "终止上市公司"}


In [39]:
ak.stock_info_sz_delist(symbol="终止上市公司")

,证券代码,证券简称,上市日期,终止上市日期
0,000003,PT金田Ａ,1991-01-14,2002-06-14
1,000005,ST星源,1990-12-10,2024-04-26
2,000013,*ST石化A,1992-05-06,2004-09-20
3,000015,PT中浩Ａ,1992-06-25,2001-10-25
4,000018,神城A退,1992-06-16,2020-01-07
...,...,...,...,...
190,300495,*ST美尚,2015-12-22,2024-09-20
191,300526,中潜退,2016-08-02,2023-07-12
192,300630,普利退,2017-03-28,2025-05-22
193,300742,越博退,2018-05-08,2024-07-22


两网及退市

In [40]:
ak.stock_staq_net_stop()

,序号,代码,名称
0,1,420280,锦州港B3
1,2,420273,凌云B5
2,3,420254,R旭电B1
3,4,420244,建车B3
4,5,420223,凯马B5
...,...,...,...
95,96,400208,金鹤5
96,97,400207,R鸿达1
97,98,400206,爱迪3
98,99,400205,泛海3


### 股票更名

In [43]:
ak.stock_info_change_name(symbol="600008")

,index,name
0,1,G首创
1,2,首创股份
2,3,首创环保


* 深证
* symbol="全称变更"; choice of {"全称变更", "简称变更"}


In [44]:
ak.stock_info_sz_change_name(symbol="全称变更")

,变更日期,证券代码,证券简称,变更前全称,变更后全称
0,1992-02-28,000004,*ST国华,蛇口安达运输股份有限公司,深圳市蛇口安达实业股份有限公司
1,1993-03-08,200012,南 玻Ｂ,中国南方玻璃股份有限公司,中国南玻集团股份有限公司
2,1993-03-08,000012,南 玻Ａ,中国南方玻璃股份有限公司,中国南玻集团股份有限公司
3,1993-07-12,000009,中国宝安,深圳市宝安企业(集团)股份有限公司,中国宝安集团股份有限公司
4,1993-12-20,000005,ST星源,深圳原野实业股份有限公司,深圳世纪星源股份有限公司
...,...,...,...,...,...
1687,2025-07-21,000652,泰达股份,天津泰达股份有限公司,天津泰达资源循环集团股份有限公司
1688,2025-07-23,000722,湖南发展,湖南发展集团股份有限公司,湖南能源集团发展股份有限公司
1689,2025-07-23,000506,招金黄金,中润资源投资股份有限公司,招金国际黄金股份有限公司
1690,2025-07-30,000595,*ST宝实,宝塔实业股份有限公司,宁夏国运新能源股份有限公司


symbol="证监会行业分类"; choice of {"证监会行业分类", "国证行业分类"}

In [45]:
ak.stock_industry_pe_ratio_cninfo(symbol="证监会行业分类", date="20250212")

,变动日期,行业分类,行业层级,行业编码,行业名称,公司数量,纳入计算公司数量,总市值-静态,净利润-静态,静态市盈率-加权平均,静态市盈率-中位数,静态市盈率-算术平均
0,2025-02-12,中国上市公司协会上市公司行业分类标准,1.0,A,农、林、牧、渔业,46.0,26.0,1908.5308,164.7542,11.58,36.52,60.41
1,2025-02-12,中国上市公司协会上市公司行业分类标准,2.0,A01,农业,10.0,8.0,648.5683,17.9204,36.19,42.49,56.45
2,2025-02-12,中国上市公司协会上市公司行业分类标准,2.0,A02,林业,2.0,2.0,38.0012,1.9804,19.19,127.76,127.76
3,2025-02-12,中国上市公司协会上市公司行业分类标准,2.0,A03,畜牧业,20.0,5.0,628.4376,115.7393,5.43,16.35,13.70
4,2025-02-12,中国上市公司协会上市公司行业分类标准,2.0,A04,渔业,6.0,4.0,134.4028,19.5600,6.87,40.95,98.23
...,...,...,...,...,...,...,...,...,...,...,...,...
115,2025-02-12,中国上市公司协会上市公司行业分类标准,2.0,R88,文化艺术业,3.0,2.0,117.5700,4.1684,28.21,31.58,31.58
116,2025-02-12,中国上市公司协会上市公司行业分类标准,2.0,R89,体育,2.0,1.0,77.4327,0.7195,107.62,107.62,107.62
117,2025-02-12,中国上市公司协会上市公司行业分类标准,2.0,R90,娱乐业,NaN,NaN,NaN,NaN,NaN,NaN,NaN
118,2025-02-12,中国上市公司协会上市公司行业分类标准,1.0,S,综合,6.0,4.0,242.1529,2.0048,120.78,126.70,128.76


In [46]:
ak.stock_hold_num_cninfo(date="20240930")

,证券代码,证券简称,变动日期,本期股东人数,上期股东人数,股东人数增幅,本期人均持股数量,上期人均持股数量,人均持股数量增幅
0,002054,德美化工,2024-09-30,19459.0,19814.0,-1.79,24776.0,24332.0,1.82
1,002055,得润电子,2024-09-30,69999.0,76707.0,-8.74,8636.0,7881.0,9.58
2,002056,横店东磁,2024-09-30,85607.0,89086.0,-3.91,19002.0,18260.0,4.06
3,600048,保利发展,2024-09-30,248055.0,165215.0,50.14,48257.0,72454.0,-33.40
4,002057,中钢天源,2024-09-30,39604.0,40145.0,-1.35,19152.0,18908.0,1.29
...,...,...,...,...,...,...,...,...,...
5130,600918,中泰证券,2024-09-30,119428.0,120102.0,-0.56,58350.0,58023.0,0.56
5131,601375,中原证券,2024-09-30,123202.0,119690.0,2.93,37685.0,38791.0,-2.85
5132,601108,财通证券,2024-09-30,130460.0,127288.0,2.49,35595.0,36482.0,-2.43
5133,002945,华林证券,2024-09-30,56880.0,52880.0,7.56,47468.0,51059.0,-7.03


实际控制人持股变动

symbol="全部"; choice of {"单独控制", "实际控制人", "一致行动人", "家族控制", "全部"}; 从 2010 开始


In [ ]:
ak.stock_hold_control_cninfo(symbol="全部")

对外担保 symbol="全部"; choice of {"全部", "深市主板", "沪市", "创业板", "科创板"}



In [ ]:
ak.stock_cg_guarantee_cninfo(symbol="全部", start_date="20240601", end_date="20250212")

股权质押

In [ ]:
ak.stock_cg_equity_mortgage_cninfo(date="20250212")

A 股个股指标
symbol="000001"; 参见 ak.stock_a_indicator_lg(symbol="all") 获取股票代码

pe	市盈率
pe_ttm	市盈率TTM
pb		市净率
ps	市销率
ps_ttm	市销率TTM
dv_ratio		股息率
dv_ttm		股息率TTM
total_mv		总市值

In [ ]:
ak.stock_a_indicator_lg(symbol="600409")

A 股股息率 symbol="上证A股"; choice of {"上证A股", "深证A股", "创业板", "科创板"}


In [ ]:
ak.stock_a_gxl_lg(symbol="深证A股")

大盘拥挤度

In [ ]:
ak.stock_a_congestion_lg()

股债利差

In [ ]:
ak.stock_ebs_lg()

A 股等权重与中位数市盈率

In [ ]:
ak.stock_a_ttm_lyr()

A 股等权重与中位数市净率

In [ ]:
ak.stock_a_all_pb()

主板市盈率 symbol="上证"; choice of {"上证", "深证", "创业板", "科创版"}


In [ ]:
ak.stock_market_pe_lg(symbol="上证")

指数市盈率symbol="上证50"; choice of {"上证50", "沪深300", "上证380", "创业板50", "中证500", "上证180", "深证红利", "深证100", "中证1000", "上证红利", "中证100", "中证800"}


In [ ]:
ak.stock_index_pe_lg(symbol="上证50")

主板市净率 symbol="上证"; choice of {"上证", "深证", "创业板", "科创版"}


In [ ]:
ak.stock_market_pb_lg(symbol="上证")

In [ ]:
ak.stock_index_pb_lg(symbol="上证50")

个股估值

In [ ]:
ak.stock_value_em(symbol="300766")

创新高和新低的股票数量 symbol="all"; {"all": "全部A股", "sz50": "上证50", "hs300": "沪深300", "zz500": "中证500"}


In [ ]:
ak.stock_a_high_low_statistics(symbol="zz500")

破净股统计 symbol="全部A股"; choice of {"全部A股", "沪深300", "上证50", "中证500"}


In [ ]:
ak.stock_a_below_net_asset_statistics(symbol="全部A股")

In [ ]:
ak.stock_report_fund_hold(symbol="基金持仓", date="20241231")

机构买卖每日统计 
机构席位追踪 symbol="近一月"; choice of {"近一月", "近三月", "近六月", "近一年"}

In [ ]:
ak.stock_lhb_jgmmtj_em(start_date="20250101", end_date="20250212")

In [ ]:
ak.stock_lhb_jgstatistic_em(symbol="近一月")

活跃 A 股统计 symbol='近三月'; choice of {'近一月', '近三月', '近六月', '近一年'}

In [ ]:
ak.stock_dzjy_hygtj(symbol='近三月')

融资融券汇总

标的证券名单及保证金比例查询

In [ ]:
ak.stock_margin_ratio_pa(date="20250213")

In [ ]:
ak.stock_margin_account_info()

In [ ]:
ak.stock_margin_sse(start_date="20250212", end_date="20250212")

In [ ]:
ak.stock_margin_szse(date="20250212")

In [ ]:
ak.stock_margin_detail_sse(date="20250212")

In [ ]:
ak.stock_margin_detail_szse(date="20250212")

In [ ]:
ak.stock_margin_underlying_info_szse(date="20250212")

盈利预测-东方财富  symbol="", 默认为获取全部数据; symbol="船舶制造", 则获取具体行业板块的数据; 行业板块可以通过 ak.stock_board_industry_name_em() 接口获取


In [ ]:
ak.stock_profit_forecast_em()

indicator="预测年报每股收益"; choice of {"预测年报每股收益", "预测年报净利润", "业绩预测详表-机构", "业绩预测详表-详细指标预测"}


In [ ]:
ak.stock_profit_forecast_ths(symbol="600519", indicator="预测年报每股收益")

### 概念板块
* 同花顺

In [47]:
ak.stock_board_concept_name_ths()

  0%|          | 0/40 [00:00<?, ?it/s]

,name,code
0,阿尔茨海默概念,308614
1,AI PC,309121
2,AI手机,309120
3,AI语料,309126
4,阿里巴巴概念,301558
...,...,...
378,AI眼镜,309152
379,光纤概念,309151
380,同花顺果指数,309136
381,维生素,309135


* 东方财富

In [48]:
ak.stock_board_concept_name_em()

  0%|          | 0/4 [00:00<?, ?it/s]

,排名,板块名称,板块代码,最新价,涨跌额,涨跌幅,总市值,换手率,上涨家数,下跌家数,领涨股票,领涨股票-涨跌幅
0,1,辅助生殖,BK0939,1141.98,25.83,2.31,593503392000,2.89,35,19,利德曼,19.93
1,2,昨日连板_含一字,BK1051,66278.21,1445.87,2.23,92404206000,9.09,8,5,安正时尚,10.02
2,3,国资云概念,BK1008,1647.25,34.41,2.13,404454432000,2.17,12,6,数字认证,20.00
3,4,虚拟机器人,BK1190,1265.47,26.16,2.11,648028352000,1.99,7,6,易点天下,18.64
4,5,液冷概念,BK1138,1303.80,26.98,2.11,1269925600000,2.33,54,19,英维克,10.01
...,...,...,...,...,...,...,...,...,...,...,...,...
432,433,可燃冰,BK0818,1350.85,-34.90,-2.52,235555720000,0.63,0,9,纽威股份,-0.72
433,434,水产养殖,BK0967,932.26,-24.74,-2.59,264013610000,2.29,2,15,海大集团,1.41
434,435,稀缺资源,BK0519,24658.20,-701.86,-2.77,3306824928000,1.02,2,77,豫光金铅,1.36
435,436,昨日触板,BK0817,59.73,-1.77,-2.88,308037712000,4.04,4,18,华润双鹤,1.93


东方财富-成份股 symbol="车联网"; 可以通过调用 ak.stock_board_concept_name_em() 查看东方财富-概念板块的所有行业名称


In [49]:
ak.stock_board_concept_cons_em(symbol="高带宽内存")

  0%|          | 0/4 [00:00<?, ?it/s]

0it [00:00, ?it/s]

,序号,代码,名称,最新价,涨跌幅,涨跌额,成交量,成交额,振幅,最高,最低,今开,昨收,换手率,市盈率-动态,市净率
0,1,688627,精智达,94.68,5.06,4.56,20738,1.951744e+08,6.92,95.50,89.26,89.26,90.12,2.87,-137.23,5.39
1,2,002436,兴森科技,19.39,4.30,0.80,1593079,3.100398e+09,5.65,20.00,18.95,19.27,18.59,10.55,879.08,6.51
2,3,688535,华海诚科,81.46,2.21,1.76,20571,1.680664e+08,4.77,82.81,79.01,79.70,79.70,3.92,227.97,6.39
3,4,603283,赛腾股份,35.05,2.07,0.71,116662,4.143768e+08,5.74,36.12,34.15,34.15,34.34,4.29,35.71,3.15
4,5,688262,国芯科技,27.87,0.50,0.14,55219,1.545916e+08,2.06,28.22,27.65,27.80,27.73,1.64,-67.37,4.34
5,6,300475,香农芯创,34.14,0.41,0.14,83692,2.889369e+08,3.68,35.10,33.85,34.00,34.00,1.87,236.62,5.21
6,7,688035,德邦科技,42.30,0.19,0.08,11799,5.040396e+07,2.34,43.19,42.20,42.30,42.22,1.33,55.42,2.63
7,8,603005,晶方科技,28.80,0.17,0.05,130944,3.794090e+08,1.91,29.15,28.60,28.61,28.75,2.01,71.85,4.38
8,9,688167,炬光科技,84.45,0.01,0.01,15338,1.312053e+08,3.28,86.60,83.83,84.09,84.44,1.71,-59.37,3.61
9,10,838971,天马新材,35.74,-0.31,-0.11,10848,3.886678e+07,1.79,36.14,35.50,35.64,35.85,1.29,966.10,8.66


东方财富-指数 symbol="长寿药"; 可以通过调用 ak.stock_board_concept_name_em() 查看东方财富-概念板块的所有概念代码
 
 period="5"; choice of {"1", "5", "15", "30", "60"}


In [ ]:
ak.stock_board_concept_hist_em(symbol="高带宽内存", period="daily", start_date="20220101", end_date="20250212", adjust="")

### 行业板块
* 同花顺

In [50]:
ak.stock_board_industry_summary_ths()

  0%|          | 0/2 [00:00<?, ?it/s]

,序号,板块,涨跌幅,总成交量,总成交额,净流入,上涨家数,下跌家数,均价,领涨股,领涨股-最新价,领涨股-涨跌幅
0,1,通信设备,1.24,1257.82,479.51,26.88,57,29,38.12,鼎通科技,96.96,11.46
1,2,软件开发,1.09,1598.31,353.36,28.25,90,45,22.10,数字认证,36.60,20.00
2,3,生物制品,0.92,518.51,116.22,3.68,23,30,22.42,安科生物,11.54,14.37
3,4,通信服务,0.89,596.66,74.79,1.63,34,11,12.53,中嘉博创,3.85,10.00
4,5,化学制药,0.82,2807.75,411.78,-3.11,67,86,14.66,南新制药,17.14,20.03
...,...,...,...,...,...,...,...,...,...,...,...,...
85,86,能源金属,-2.48,173.71,56.15,-5.03,0,13,32.32,博迁新材,41.40,-0.31
86,87,贵金属,-2.63,243.72,34.16,-9.02,0,11,14.02,恒邦股份,11.59,-2.03
87,88,油气开采及服务,-2.98,571.72,36.19,-10.71,0,20,6.33,中曼石油,19.54,-1.11
88,89,煤炭开采加工,-3.34,865.09,66.74,-18.25,0,34,7.72,安源煤业,6.11,-1.45


* 东方财富

In [51]:
ak.stock_board_industry_name_em()

0it [00:00, ?it/s]

,排名,板块名称,板块代码,最新价,涨跌额,涨跌幅,总市值,换手率,上涨家数,下跌家数,领涨股票,领涨股票-涨跌幅
0,1,通信设备,BK0448,16488.96,194.35,1.19,2001257264000,2.74,74,30,长飞光纤,10.00
1,2,生物制品,BK1044,1174.77,13.33,1.15,1644837568000,1.74,41,32,安科生物,14.27
2,3,软件开发,BK0737,1256.18,14.13,1.14,2518282112000,2.16,133,59,数字认证,20.00
3,4,化学制药,BK0465,40755.13,456.23,1.13,2295745344000,2.89,71,70,汉商集团,10.00
4,5,互联网服务,BK0447,29567.06,324.18,1.11,1878571696000,2.31,98,44,易点天下,19.14
...,...,...,...,...,...,...,...,...,...,...,...,...
81,82,贵金属,BK0732,1439.73,-38.91,-2.63,388317504000,1.26,0,11,四川黄金,-1.99
82,83,能源金属,BK1015,533.58,-14.99,-2.73,377225440000,1.70,0,13,威领股份,-0.66
83,84,采掘行业,BK1017,1382.91,-39.75,-2.79,314364880000,1.01,1,20,广东宏大,0.14
84,85,煤炭行业,BK0437,10859.36,-367.77,-3.28,1735617600000,0.81,0,35,伊泰Ｂ股,-0.71


东方财富-成份股

In [52]:
ak.stock_board_industry_cons_em(symbol="电机")

0it [00:00, ?it/s]

0it [00:00, ?it/s]

,序号,代码,名称,最新价,涨跌幅,涨跌额,成交量,成交额,振幅,最高,最低,今开,昨收,换手率,市盈率-动态,市净率
0,1,300660,江苏雷利,46.79,2.39,1.09,119072,5.563438e+08,4.40,47.33,45.32,45.32,45.70,2.67,55.88,6.21
1,2,603728,鸣志电器,59.84,2.17,1.27,53125,3.170450e+08,3.67,60.44,58.29,58.30,58.57,1.27,871.64,8.60
2,3,003021,兆威机电,113.90,1.51,1.69,37024,4.232590e+08,3.45,115.43,111.56,111.79,112.21,1.80,125.02,8.48
3,4,600416,湘电股份,14.03,0.79,0.11,259500,3.692924e+08,4.31,14.59,13.99,14.08,13.92,1.96,49.34,2.42
4,5,002801,微光股份,33.58,0.66,0.22,20517,6.921022e+07,2.46,33.92,33.10,33.10,33.36,1.77,22.36,4.40
5,6,600580,卧龙电驱,23.15,0.65,0.15,387481,9.005053e+08,2.70,23.46,22.84,22.84,23.00,2.48,33.72,3.56
6,7,301226,祥明智能,27.27,0.52,0.14,8491,2.330191e+07,1.99,27.59,27.05,27.07,27.13,0.78,104.34,3.30
7,8,873593,鼎智科技,37.04,0.41,0.15,23052,8.537619e+07,1.63,37.25,36.65,36.77,36.89,1.43,139.20,10.21
8,9,002576,通达动力,16.60,-0.06,-0.01,24974,4.179584e+07,1.57,16.84,16.58,16.60,16.61,1.54,36.07,2.17
9,10,603350,安乃达,36.98,-0.14,-0.05,9385,3.489602e+07,1.89,37.60,36.90,37.11,37.03,2.13,34.18,3.36


In [ ]:
ak.stock_board_industry_hist_em(symbol="小金属", start_date="20211201", end_date="20250212", period="日k", adjust="")

同花顺  行业板块

In [ ]:
ak.stock_board_industry_summary_ths()

In [ ]:
ak.stock_board_industry_index_ths(symbol="元件", start_date="20240101", end_date="20250212")

股票热度 关注 讨论 交易

关注排行榜 symbol="最热门"; choice of {"本周新增", "最热门"}


In [ ]:
ak.stock_hot_follow_xq(symbol="最热门")

In [ ]:
ak.stock_hot_tweet_xq(symbol="最热门")

In [ ]:
ak.stock_hot_deal_xq(symbol="最热门")

问财

In [ ]:
ak.stock_hot_rank_wc(date="20250212")

东财 单次返回当前交易日前 100 个股票的人气排名数据

In [ ]:
ak.stock_hot_rank_em()

飙升榜-A股

In [ ]:
ak.stock_hot_up_em()

历史趋势及粉丝特征

In [ ]:
ak.stock_hot_rank_detail_em(symbol="SH600996")

热门关键词

In [ ]:
ak.stock_hot_keyword_em(symbol="SH600996")

内部交易

In [ ]:
ak.stock_inner_trade_xq()

热搜股票 time="今日"; choice of {"今日", "1小时"}

In [ ]:
ak.stock_hot_search_baidu(symbol="A股", date="20240213", time="今日")

强势股池 

In [ ]:
ak.stock_zt_pool_strong_em(date='20250206')

In [ ]:
ak.stock_zt_pool_sub_new_em(date='20250123')

In [ ]:
ak.stock_info_cjzc_em()

In [ ]:
ak.stock_info_global_em()

In [ ]:
ak.stock_info_global_sina()

In [ ]:
ak.stock_info_global_futu()

In [ ]:
ak.stock_info_global_ths()

In [ ]:
ak.stock_info_global_cls(symbol="全部")

技术指标 symbol="创月新高"; choice of {"创月新高", "半年新高", "一年新高", "历史新高"}

创新高cxg 创新低cxd 

连续上涨lxsz 连续下跌lxxd 持续放量cxfl 持续缩量cxsl 向上突破xstp 向下突破xxtp 量价齐升ljqs 量价齐跌ljqd 险资举牌xzjp

In [ ]:
ak.stock_rank_cxg_ths(symbol="创月新高")

In [ ]:
ak.stock_rank_cxd_ths(symbol="创月新低")

In [ ]:
ak.stock_rank_lxsz_ths()

In [ ]:
ak.stock_rank_cxfl_ths()

In [ ]:
ak.stock_rank_cxsl_ths()

In [ ]:
ak.stock_rank_xzjp_ths()

ESG评级

In [ ]:
ak.stock_esg_rate_sina()

In [ ]:
ak.stock_esg_msci_sina()

In [ ]:
ak.stock_esg_rft_sina()

In [ ]:
ak.stock_esg_zd_sina()

In [ ]:
ak.stock_esg_hz_sina()

A股股票指数数据 symbol="上证系列指数"；choice of {"沪深重要指数", "上证系列指数", "深证系列指数", "指数成份", "中证系列指数"}


In [ ]:
ak.stock_zh_index_spot_em(symbol="中证系列指数")

历史行情数据-东方财富 period="daily"; choice of {'daily', 'weekly', 'monthly'}

In [ ]:
ak.index_zh_a_hist(symbol="931144", period="daily", start_date="20250212", end_date="20250212")

中国股票指数成份 

In [ ]:
ak.index_stock_info()

In [ ]:
ak.index_stock_cons(symbol="000300")

In [ ]:
ak.index_stock_cons_csindex(symbol="000300")

In [ ]:
ak.sw_index_first_info()

In [ ]:
ak.sw_index_second_info()

In [ ]:
ak.sw_index_third_info()

申万三级行业成份 symbol="850111.SI"; 行业代码; 可以通过 ak.sw_index_third_info() 获取所有行业代码


In [ ]:
ak.sw_index_third_cons(symbol="859822.SI")

In [ ]:
ak.drewry_wci_index(symbol="composite")

In [ ]:
ak.index_neaw_cx()

汽车销量排行

In [ ]:
ak.car_market_cate_cpca(symbol="占比", indicator="零售")

In [ ]:
ak.car_market_cate_cpca(symbol="轿车", indicator="零售")

盖世研究院 symbol="车型榜"; choice of {"车企榜", "品牌榜", "车型榜"}

In [ ]:
ak.car_sale_rank_gasgoo(symbol="车企榜", date="202412")

In [ ]:
ak.car_sale_rank_gasgoo(symbol="品牌榜", date="202412")

In [ ]:
ak.car_sale_rank_gasgoo(symbol="车型榜", date="202412")

电影票房

In [ ]:
ak.movie_boxoffice_realtime()

In [ ]:
ak.movie_boxoffice_daily(date="20250201")

In [ ]:
ak.movie_boxoffice_yearly(date="2025")

In [ ]:
ak.video_tv()

In [ ]:
ak.video_variety_show()

In [ ]:
ak.business_value_artist()

In [ ]:
ak.online_value_artist()

微博舆情报告 time_period="CNHOUR12"

CNHOUR2	2小时
CNHOUR6	6小时
CNHOUR12	12小时
CNHOUR24	1天
CNDAY7	1周
CNDAY30	1月

In [ ]:
ak.stock_js_weibo_report(time_period="CNHOUR24")

In [ ]:
df = ak.index_news_sentiment_scope()

In [ ]:
df

In [ ]:
import plotly.express as px

In [ ]:
fig = px.line(df.tail(100), x='日期', y='市场情绪指数',)
fig.show()

In [ ]:
fig = px.line(df.tail(100), x='日期', y='沪深300指数',)
fig.show()

In [ ]:
ak.index_analysis_weekly_sw(symbol="市场表征", date="20250228")

In [ ]:
dfg = ak.macro_cons_gold()

In [ ]:
from datetime import datetime, date

dfgg = dfg[dfg['日期'] >= datetime.strptime('2016-12-19',  "%Y-%m-%d").date()]

In [ ]:
fig = px.line(dfgg, x='日期', y='总库存',)
fig.show()

In [ ]:
dfd = ak.spot_hist_sge(symbol='Au99.99')

In [ ]:
fig = px.line(dfd, x='date', y='close',)
fig.show()